processing/src/test/java/org/apache/druid/concurrent/ExecsTest.java

```diff
@@ -118,6 +122,38 @@ public void run()
     producer.shutdown();
   }
 
+  @Test
+  public void testSynchronousQueueSingleThreadedExecutor() throws Exception
+  {
+    // The implementation of Execs.newBlockingSingleThreaded() rejectedExecutionHandler should not add tasks when it's in shutDown state
+    // When capacity is 0, a SynchronousQueue is used and if a task is put in it in ShutDown state, it will forever stuck in WAITING state
+    // as executor will not take() the task to schedule it.
+    final ListeningExecutorService intermediateTempExecutor = MoreExecutors.listeningDecorator(
+        Execs.newBlockingSingleThreaded("[TASK_ID]-appenderator-abandon", 0)
+    );
+    Callable<Void> task = () -> {
+      try {
+        Thread.sleep(1000); // Simulate long-running task
+      }
+      catch (InterruptedException e) {
+        Thread.sleep(1500);
+        Thread.currentThread().interrupt(); // Restore interrupted status
+      }
+      return null;
+    };
+
+    // Submit multiple tasks together
+    intermediateTempExecutor.submit(task);
+    intermediateTempExecutor.submit(task);
+    intermediateTempExecutor.submit(task);
+
+    intermediateTempExecutor.shutdownNow();
+    // Submit task after shutDown / shutDownNow should not be added in queue
+    intermediateTempExecutor.submit(task);
+    intermediateTempExecutor.awaitTermination(10, TimeUnit.SECONDS);
+    Assert.assertTrue(intermediateTempExecutor.isShutdown());
```

Comment:
if you remove the fix to the product code and run this test, does line 152 get stuck forever?

Reply:
Yes